SU-HSDO/suhumsci
# FourKitchens Release: Sprint 11 (part 2)

# READY FOR REVIEW

## Summary
- SHS-92: Implements Codeception test for the Spotlight Slider component
- SHS-91: Implements Codeception test for Video Embeds (with captions) component
- SHS-64: Implements Codeception test for the mega menu

## Need Review By (Date)
- 11/2

## Urgency
low

## Steps to Test
1. Codeception test for Spotlight Slider
    - Verify CircleCI runs each step of the test and it passes (see output logs in CI)
    - Verify the steps defined in `FlexiblePageCest.php` properly add this component to a flexible page, fill out the necessary fields, save the component/page, and can view it on the frontend
5. Codeception test for Video Embeds
    - Verify CircleCI runs each step of the test and it passes (see output logs in CI)
    - Review the steps defined in `VideoEmbedCest.php` and verify they add a textarea component to a new page, use the entity_embed button in the ckeditor, add a new video with a custom caption, and confirm the video is properly embedded into the editor.
6. Codeception test for Mega Menu
    - Verify CircleCI runs each step of the test and it passes (see output logs in CI)
    - Verify the steps defined in `MegaMenuCest.php` properly add some test menu links, enable the mega menu, verify the mega menu is present at desktop, verify the mega menu can be toggled open on mobile, and disables the mega menu for further tests

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)
